iotilecore/iotile/core/hw/auth/cli_auth_provider.py

```diff
@@ -37,17 +38,24 @@ def derive_key(cls, password):
         return hashlib.pbkdf2_hmac('sha256', password.encode('utf-8'), b'salt', 100000)
 
     def get_root_key(self, key_type, device_id):
-        """Prompt user for the password and derive root key from it
+        """Prompt a user for the password and derive the root key from it
 
         Args:
             key_type (int): see KnownKeyRoots
-            device_id (int): uuid of the device
+            device_id (int): the uuid of the device
 
         Returns:
             bytes: the root key
         """
         self.verify_key(key_type)
 
-        password = getpass.getpass("Please, input user password for device {} :".format(device_id))
+        password = getpass.getpass("Please, input the user password for the device {} :".format(device_id))
+        userkey = CliAuthProvider.derive_key(password)
 
-        return CliAuthProvider.derive_key(password)
+        if device_id:
+            answer = input("Would you like to save the user-key till the end of the current session? (y/n)")
```

Comment:
I think that there needs to be a way to pass this type of information without user prompts. If we rely on user input, we will not be able to use this embedded in applications, right? Unless this is already the case - does this function `get_root_key` always get called when using security, or do we support non-interactive mode?

Reply:
I think I answered my own question, that this is only present when you specifically want to do a user-mode action, but it's good to double check since I didn't fully follow all of the code paths here.

Reply:
Thank you for bringing this up. The intention is to ask the pass specifically via CLI, current implementation does not support the not-interactive mode for PasswordBasedKey